### PR TITLE
Add NCCL check for XLA

### PR DESCRIPTION
**Pull request by MattConley**
Addresses issue #29886 by replacing the if_cuda checks with if_nccl, and exporting GOOGLE_NCCL similarly to GOOGLE_CUDA and GOOGLE_TENSORRT

### CODE REVIEW

File: tensorflow/tensorflow.bzl [hunk 1]
```diff
@@ -1329,8 +1330,8 @@ def tf_gpu_library(deps = None, cuda_deps = None, copts = tf_copts(), **kwargs):
     - Both deps and cuda_deps are used as dependencies.
     - The cuda runtime is added as a dependency (if necessary).
     - The library additionally passes -DGOOGLE_CUDA=1 to the list of copts.
-    - In addition, when the library is also built with TensorRT enabled, it
-        additionally passes -DGOOGLE_TENSORRT=1 to the list of copts.
+    - In addition, when the library is also built with TensorRT or NCCL enabled,
+        -DGOOGLE_TENSORRT=1 or -DGOOGLE_NCCL=1 are passed to the list of copts.
```
Comment (thomasjoerg) [inline tensorflow/tensorflow.bzl]: Nit: Please keep "it additionally passes" in the sentence.

**Review by thomasjoerg — DISMISSED** [review]

**Comment by MattConley** [comment]
Thanks for the review, just addressed the comment, let me know if I should do anything else.

**Review by thomasjoerg — DISMISSED** [review]

**Comment by thomasjoerg** [comment]
@sanjoy This LGTM, just wanted to bring this to your attention since you proposed the fix and may want to take a look yourself.

**Review by sanjoy — COMMENTED** [review]
Looks fine to me

**Comment by MattConley** [comment]
Sounds good; does this look ready to merge in?

**Comment by tfboyd** [comment]
Looks like an internal test is failing @aaroey is taking a look.

**Review by tfboyd — DISMISSED** [review]
This looks approved LGTM so it can move forward.

**Comment by rthadur** [comment]
@MattConley can you please resolve conflicts ?

**Comment by MattConley** [comment]
@rthadur Just addressed the conflict, thanks.
@thomasjoerg Are we good to merge this in?

**Review by sanjoy — APPROVED** [review]
lgtm

**Comment by mellanox-github** [comment]
Can one of the admins verify this patch?

**Comment by rthadur** [comment]
@MattConley sorry it did not merge last time , can you please resolve conflicts again so that we can import .

**Comment by chsigg** [comment]
I've implemented this slightly differently in commit 9b9bea65159eea4c8452d7f8b702b32dcb3cea8e. I'm closing this PR. Feel free to reopen if it doesn't work as expected.

**Pull request closed without merging**